plugins/env/app/models/environment_variable.rb

```diff
@@ -73,11 +73,24 @@ def config_service_location(project, deploy_group, display:)
 
     private
 
-    def env_vars_from_db(deploy, deploy_group)
+    def env_vars_from_db(deploy, deploy_group, project_specific: nil)
+      # Project Specific:
+      # nil   => project env + groups env
+      # true  => project env
+      # false => groups env
+      project_envs =
+        if project_specific.to_s == "true"
+          deploy.project.environment_variables
+        elsif project_specific.to_s == "false"
+          deploy.project.environment_variable_groups.flat_map(&:environment_variables)
```

Comment:
this is duplicating the logic from `nested_environment_variables` ... so prefer moving project_specific there

```
    def env_vars_from_db(deploy, deploy_group, **args)

deploy.project.nested_environment_variables **args

def nested_environment_variables(project_specific: nil)
```